storage/src/vespa/storage/persistence/mergehandler.cpp

```diff
@@ -617,9 +607,12 @@ MergeHandler::applyDiffLocally(
         LOG(spam, "ApplyBucketDiff(%s): Adding slot %s",
             bucket.toString().c_str(), e.toString().c_str());
 
-        applyDiffEntry(bucket, e, context, repo);
+        async_results.push_back(applyDiffEntry(bucket, e, context, repo));
         byteCount += e._headerBlob.size() + e._bodyBlob.size();
     }
+    for (auto &result_to_check : async_results) {
+        result_to_check.check_result();
```

Comment:
Should we split this into two phases (wait phase and check_result phase) to ensure all async ops have completed before potentially throwing an exception out of this method?